aiohttp/client_ws.py

```diff
@@ -377,12 +377,20 @@ async def receive(self, timeout: Optional[float] = None) -> WSMessage:
 
     async def receive_str(self, *, timeout: Optional[float] = None) -> str:
         msg = await self.receive(timeout)
+        if msg.type not in MESSAGE_TYPES_WITH_CONTENT:
+            raise RuntimeError(
+                f"Received message {msg.type}:{msg.data!r} has no content"
+            )
         if msg.type is not WSMsgType.TEXT:
             raise TypeError(f"Received message {msg.type}:{msg.data!r} is not str")
         return cast(str, msg.data)
 
     async def receive_bytes(self, *, timeout: Optional[float] = None) -> bytes:
         msg = await self.receive(timeout)
+        if msg.type not in MESSAGE_TYPES_WITH_CONTENT:
```

Comment:
This check can be moved inside the `if msg.type is not WSMsgType.BINARY:` block so we only check it if we know we are going to raise